### ISSUE TITLE
[FalsePositive][SuspiciousAssignmentsInspection] `null` destructuring

### ISSUE THREAD

**Issue by kaznovac** [kalessil/phpinspectionsea#2019]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 6.3.0 |
| Language level | 8.4                       |

Current behaviour
--
minimal reproducer, both are reported as `assigned value isn't an array`
```php
[$a, $b] = null;
list($c, $d) = null;
```

Expected behaviour
--
those are legitimate php constructs, should not be flagged as errors
e.g. https://3v4l.org/Utm3F#v
